staging/src/k8s.io/client-go/util/flowcontrol/backoff_test.go

```diff
@@ -250,3 +251,75 @@ func TestBackoffWithJitter(t *testing.T) {
 
 	t.Logf("exponentially backed off jittered delays: %v", delays)
 }
+
+func TestIsInBackOffSinceUpdateTick(t *testing.T) {
+	id := "_idIsInBackOffSinceUpdate"
+	tc := testingclock.NewFakeClock(time.Now())
+	step := time.Second
+	maxDuration := 300 * step
+	initial := 10 * step
+	b := NewFakeBackOff(initial, maxDuration, tc)
+	startTime := tc.Now()
+
+	want := map[int]time.Duration{
+		0:    0 * time.Second,
+		10:   10 * time.Second,
+		30:   20 * time.Second,
+		70:   40 * time.Second,
+		150:  80 * time.Second,
+		310:  160 * time.Second,
+		610:  300 * time.Second,
+		910:  300 * time.Second,
+		1210: 300 * time.Second,
+		1510: 300 * time.Second,
+		1810: 300 * time.Second,
+	}
+	got := map[int]time.Duration{}
+	for i := 0; i != 2000; i++ {
+		tc.SetTime(startTime.Add(time.Duration(i) * step))
+		inBackOff := b.IsInBackOffSinceUpdate(id, tc.Now())
+		if !inBackOff {
+			got[i] = b.Get(id)
+			b.Next(id, tc.Now())
+		}
+	}
+
+	if !reflect.DeepEqual(want, got) {
+		t.Errorf("expected \n%v got \n%v", want, got)
+	}
+}
+
+func TestIsInBackOffSinceTick(t *testing.T) {
+	id := "_idIsInBackOffSince"
+	tc := testingclock.NewFakeClock(time.Now())
+	step := time.Second
+	maxDuration := 300 * step
+	initial := 10 * step
+	b := NewFakeBackOff(initial, maxDuration, tc)
+	startTime := tc.Now()
+
+	want := map[int]time.Duration{
+		0:    0 * time.Second,
+		601:  10 * time.Second,
+		1202: 10 * time.Second,
+		1803: 10 * time.Second,
+		2404: 10 * time.Second,
+		3005: 10 * time.Second,
+		3606: 10 * time.Second,
+		4207: 10 * time.Second,
+		4808: 10 * time.Second,
+	}
+	got := map[int]time.Duration{}
+	for i := 0; i != 5000; i++ {
+		tc.SetTime(startTime.Add(time.Duration(i) * step))
+		inBackOff := b.IsInBackOffSince(id, tc.Now())
```

Comment:
I don't think IsInBackOffSince() is expected to be called with eventTime always set to tc.Now(). 
I was expecting a test where eventTime remains the same, and `IsInBackOffSince` returns different results as the fakeClock advances.

I understand that you want to verify that IsInBackOffSince returns different results based on if `eventTime-lastUpdate > 2*maxDuration`. I would test that in a separate test, maybe `TestIsInBackOffSinceExpiration`.

cc @tkashem who recently touched this package.

Reply:
The only place where this algorithm is currently used is the container restart strategy. I refer that to the test there.
Since the container restart strategy is not an exponential backoff, I want to try to prove that another algorithm should be used instead.
#103257